include/envoy/network/connection.h

```diff
@@ -194,7 +194,7 @@ class Connection : public Event::DeferredDeletable, public FilterManager {
   /**
    * @return State the current state of the connection.
    */
-  virtual State state() const PURE;
```

Comment:
I think this is still functionally const, yes?  I think I'd prefer const and non-const ioHandle() accessors to changing the functions to look mutable